cpp/JSIUtils/MGLJSIUtils.h

```diff
@@ -21,4 +22,12 @@ inline bool CheckSizeInt32(jsi::Runtime &runtime, jsi::ArrayBuffer &buffer) {
   return buffer.size(runtime) <= INT_MAX;
 }
 
+inline bool CheckIsInt32(const jsi::Value &value) {
+  if (!value.isNumber()) {
+    return false;
+  }
+  double d = value.asNumber();
```

Comment:
I think `get...` is faster if you already did a `is...` check before. `as...` is only useful if you don't want the `is...` check before and just throw if not met. `get..` will assert, which can't happen since we already checked `is...`.